payu/runlog.py

```diff
@@ -274,3 +274,29 @@ def authenticate(self):
 
         github_auth = (github_username, github_password)
         return github_auth
+
+
+# Some git utility functions
+
+def commit_hash(dir='.'):
+    """
+    Return commit hash for HEAD of checked out branch of the
+    specified directory.
+    """
+
+    cmd = ['git', 'rev-parse', 'HEAD']
+
+    try:
+        with open(os.devnull, 'w') as devnull:
+            revision_hash = subprocess.check_output(
+              cmd,
+              cwd=dir,
+              stderr=devnull
+            )
+        if sys.version_info.major > 2:
```

Comment:
Should not need an explicit version check, should be ok to do `.decode('ascii`)` regardless of Py 2 vs 3.

Reply:
Subprocess `check_output` returns strings in python2 and byte objects in python3. If we do `.decode('ascii')` in python2 we end up with a `unicode` type, which can be problematic. We want it to be a string.

Happy to defer, I hate this stuff, but I'm pretty sure I had to do it this way.

Reply:
Oh huh, you're right, I did not know that.  But it will still be a unicode string in 3 anyway, right?

Reply:
I mean, we are talking about hashes anyway, so should be ASCII regardless, right?

Reply:
For python3 the `.decode('ascii')` will transform the byte object to a str (not unicode, python3 no longer makes that distinction). python2 returns a string, so doesn't need the `decode` step.

Reply:
Like I said, I hate this stuff. Most people don't bother supporting both, which is where the problem lies.

Reply:
I had thought a Python 3 `str` is actually a unicode string, but I guess it's not that important.

Once upon a time I had come up with some way to deal with this that didn't involve version checks, but it's probably not important, can deal with it another time.

Reply:
Could check the return type I guess. Agreed this can be tackled at another time. At some point python2 support will be dropped, and this stuff can be removed with a nice fat

```python
assert(sys.version_info.major > 2)
```

somewere, or requiring a `python3` interpreter.